From 48cd5fea4d5d962baa772d3753311912d7684380 Mon Sep 17 00:00:00 2001
From: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
Date: Mon, 7 May 2018 11:30:16 +0530
Subject: [PATCH 3854/5855] drm/amdgpu: compilation patch at move regamma from
 opp to dpp for dce Signed-off-by: Pavan Kumar Ramayanam
 <pavan.ramayanam@amd.com>

---
 drivers/gpu/drm/amd/display/dc/core/dc.c           | 30 +++-------------------
 drivers/gpu/drm/amd/display/dc/dce/dce_transform.h |  2 --
 .../amd/display/dc/dce110/dce110_hw_sequencer.c    |  4 +--
 3 files changed, 5 insertions(+), 31 deletions(-)
 mode change 100644 => 100755 drivers/gpu/drm/amd/display/dc/dce/dce_transform.h

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc.c b/drivers/gpu/drm/amd/display/dc/core/dc.c
index 99f157d..f1bbfa4 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc.c
@@ -917,30 +917,6 @@ bool dc_commit_state(struct dc *dc, struct dc_state *context)
 }
 
 
-bool dc_enable_stereo(
-	struct dc *dc,
-	struct validate_context *context,
-	const struct dc_stream *streams[],
-	uint8_t stream_count)
-{
-	bool ret = true;
-	int i, j;
-	struct pipe_ctx *pipe;
-	struct core_dc *core_dc = DC_TO_CORE(dc);
-	for (i = 0; i < MAX_PIPES; i++) {
-		if (context != NULL)
-			pipe = &context->res_ctx.pipe_ctx[i];
-		else
-			pipe = &core_dc->current_context->res_ctx.pipe_ctx[i];
-		for (j = 0 ; pipe && j < stream_count; j++)  {
-			if (streams[j] && streams[j] == &pipe->stream->public &&
-				core_dc->hwss.setup_stereo)
-				core_dc->hwss.setup_stereo(pipe, core_dc);
-		}
-	}
-	return ret;
-}
-
 bool dc_post_update_surfaces_to_stream(struct dc *dc)
 {
 	int i;
@@ -1629,7 +1605,7 @@ void dc_flip_plane_addrs(
 		plane_state->flip_immediate = flip_addrs[i].flip_immediate;
 
 		for (j = 0; j < core_dc->res_pool->pipe_count; j++) {
-			struct pipe_ctx *pipe_ctx = &core_dc->current_context->res_ctx.pipe_ctx[j];
+			struct pipe_ctx *pipe_ctx = &core_dc->current_state->res_ctx.pipe_ctx[j];
 
 			if (pipe_ctx->plane_state != plane_state)
 				continue;
@@ -1667,8 +1643,8 @@ void dc_set_power_state(
 {
         atomic_t ref_count;
 
-	core_dc->previous_power_state = core_dc->current_power_state;
-	core_dc->current_power_state = video_power_state;
+	dc->previous_power_state = dc->current_power_state;
+	dc->current_power_state = video_power_state;
 
 	switch (power_state) {
 	case DC_ACPI_CM_POWER_STATE_D0:
diff --git a/drivers/gpu/drm/amd/display/dc/dce/dce_transform.h b/drivers/gpu/drm/amd/display/dc/dce/dce_transform.h
old mode 100644
new mode 100755
index 81d5a7f..45e6c5b
--- a/drivers/gpu/drm/amd/display/dc/dce/dce_transform.h
+++ b/drivers/gpu/drm/amd/display/dc/dce/dce_transform.h
@@ -239,8 +239,6 @@
 	XFM_SF(SCL0_SCL_UPDATE, SCL_COEF_UPDATE_COMPLETE, mask_sh), \
 	XFM_SF(LB0_LB_DATA_FORMAT, ALPHA_EN, mask_sh), \
 	XFM_SF(DCFE0_DCFE_MEM_PWR_CTRL, SCL_COEFF_MEM_PWR_DIS, mask_sh), \
-        XFM_SF(DCFE0_DCFE_MEM_PWR_CTRL, DCP_REGAMMA_MEM_PWR_DIS, mask_sh),\
-        XFM_SF(DCFE0_DCFE_MEM_PWR_CTRL, DCP_LUT_MEM_PWR_DIS, mask_sh),\
 	XFM_SF(DCFE0_DCFE_MEM_PWR_STATUS, SCL_COEFF_MEM_PWR_STATE, mask_sh), \
 	XFM_SF(SCL0_SCL_MODE, SCL_PSCL_EN, mask_sh)
 
diff --git a/drivers/gpu/drm/amd/display/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/display/dc/dce110/dce110_hw_sequencer.c
index 51987c6..85d85a8 100644
--- a/drivers/gpu/drm/amd/display/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/display/dc/dce110/dce110_hw_sequencer.c
@@ -2632,7 +2632,7 @@ static void dce110_apply_ctx_for_surface(
 
 	for (i = 0; i < dc->res_pool->pipe_count; i++) {
 		struct pipe_ctx *pipe_ctx = &context->res_ctx.pipe_ctx[i];
-		struct pipe_ctx *old_pipe_ctx = &dc->current_context->res_ctx.pipe_ctx[i];
+		struct pipe_ctx *old_pipe_ctx = &dc->current_state->res_ctx.pipe_ctx[i];
 
 		if (stream == pipe_ctx->stream) {
 			if (!pipe_ctx->top_pipe &&
@@ -2666,7 +2666,7 @@ static void dce110_apply_ctx_for_surface(
 
 	for (i = 0; i < dc->res_pool->pipe_count; i++) {
 		struct pipe_ctx *pipe_ctx = &context->res_ctx.pipe_ctx[i];
-		struct pipe_ctx *old_pipe_ctx = &dc->current_context->res_ctx.pipe_ctx[i];
+		struct pipe_ctx *old_pipe_ctx = &dc->current_state->res_ctx.pipe_ctx[i];
 
 		if ((stream == pipe_ctx->stream) &&
 			(!pipe_ctx->top_pipe) &&
-- 
2.7.4

